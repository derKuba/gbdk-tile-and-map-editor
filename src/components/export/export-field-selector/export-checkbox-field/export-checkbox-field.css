
.field-box {
  display: inline-flex;
  border: 1px solid black;
}

.field {
  
}