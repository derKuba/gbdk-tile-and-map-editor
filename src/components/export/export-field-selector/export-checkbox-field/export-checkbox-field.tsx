import { Component, h, Prop } from '@stencil/core';

@Component({
    tag: 'mvjp-export-checkbox-field',
    styleUrl: 'export-checkbox-field.css',
    shadow: false,
})
export class ExportCheckboxField {
    /**
     *  the selected field of the matrix
     */
    @Prop() field: number[][];

    /**
     *  state of the checkbox
     */
    @Prop() isActive: boolean;

    /**
     *  position of the checkbox in the sprite tile viewer
     *
     */
    @Prop() position: number;

    /**
     *  event to select this specific checkbox
     */
    @Prop() setSelectedTile: Function;

    private mapColorCodeToColour = colorCode => {
        const colors: string[] = ['white', 'grey', 'grey-dark', 'black'];
        return colors[colorCode];
    };

    private handleClick = () => {
        this.setSelectedTile(this.position);
    };

    render() {
        const field = this.field.map((tileRow: number[], y: number) => (
            <div class="matrixRow" key={`index_${y}`}>
                {tileRow.map((colorCode: number, x: number) => (
                    <div
                        onClick={this.handleClick}
                        class={`field gb-${this.mapColorCodeToColour(colorCode)}`}
                        key={`index_${y}_${x}`}></div>
                ))}
            </div>
        ));

        const isActiveClass = this.isActive === true ? 'active' : '';

        return <div class={`field-box ${isActiveClass}`}>{field}</div>;
    }
}
