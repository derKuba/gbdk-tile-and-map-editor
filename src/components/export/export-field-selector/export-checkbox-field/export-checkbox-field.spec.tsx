import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { ExportCheckboxField } from './export-checkbox-field';

const field = [
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0],
];

describe('mvjp-export-checkbox-field', () => {
    it('builds', () => {
        expect(new ExportCheckboxField()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const page = await newSpecPage({
            components: [ExportCheckboxField],
            template: () => {
                return (
                    <div>
                        <mvjp-export-checkbox-field
                            field={[...field]}
                            position={0}
                            isActive={true}
                            setSelectedTile={() => {}}
                        />
                        <mvjp-export-checkbox-field
                            field={[...field]}
                            position={1}
                            isActive={false}
                            setSelectedTile={() => {}}
                        />
                    </div>
                );
            },
        });

        expect(page.root).toMatchSnapshot();
    });

    it('should handle click', async () => {
        const setSelectedTile = jest.fn();

        const page = await newSpecPage({
            components: [ExportCheckboxField],
            template: () => {
                return (
                    <mvjp-export-checkbox-field
                        field={[...field]}
                        position={23}
                        isActive={true}
                        setSelectedTile={setSelectedTile}
                    />
                );
            },
        });

        page.root.querySelector('.field').click();
        await page.waitForChanges();
        expect(setSelectedTile).toBeCalledWith(23);
    });
});
