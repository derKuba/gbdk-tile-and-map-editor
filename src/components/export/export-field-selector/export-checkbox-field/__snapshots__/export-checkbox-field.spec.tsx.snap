// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-export-checkbox-field matches snapshot 1`] = `
<mvjp-export-checkbox-field>
  <div class="active field-box">
    <div class="matrixRow">
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
    </div>
    <div class="matrixRow">
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
    </div>
    <div class="matrixRow">
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
    </div>
    <div class="matrixRow">
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
    </div>
    <div class="matrixRow">
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
    </div>
    <div class="matrixRow">
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
    </div>
    <div class="matrixRow">
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
    </div>
    <div class="matrixRow">
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
      <div class="field gb-white"></div>
    </div>
  </div>
</mvjp-export-checkbox-field>
`;
